website/client-old/js/controllers/inventoryCtrl.js

```diff
@@ -382,6 +382,24 @@ habitrpg.controller("InventoryCtrl",
       });
     };
 
+    $scope.shouldShowPremiumMountRow = function (potion) {
+      potion = Content.premiumHatchingPotions[potion].key;
+      var pets = Object.keys(user.items.pets);
+      return pets.find(function (pet) {
+        return pet.indexOf(potion) !== -1;
+      });
+    };
+
+    $scope.shouldShowPremiumMountSection = function () {
```

Comment:
The logic here should basically be if the user has ever had a premium pet, right? 

You can use a similar method to what I showed you above, asserting on the `pet.type === 'premium'` property. Let me know if you need help.